### PR TITLE
Define HTTP server

### PR DESCRIPTION
**Pull request by fvovan**
Authentication based on defined secret.

Actions:
- `/stopStream` (using clientKey/special identifier)
- `/status` (list existing streamchannels and their streams) - is it really needed?

### CODE REVIEW

**Comment by fvovan** [comment]
@vogdb please review.
I had to pull your changes from https://github.com/cargomedia/cm-janus/pull/6 to keep all things here.

File: lib/cm-http-server.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+var express = require('express');
+var http = require('http');
+var bodyParser = require('body-parser');
+var _ = require('underscore');
+
+var CMHttpServer = function() {
```
Comment (tomaszdurka) [inline lib/cm-http-server.js]: I wouldn't call it `CM...` server as it's not really CM related.

**Comment by tomaszdurka** [comment]
Can you replace all `console.log()`s with `getLog().debug()`?

**Comment by vogdb** [comment]
Also lets change `getLog().debug()` to `log.debug()`.

**Comment by tomaszdurka** [comment]
Wdyt about:
- Storing `router` and `server` on `HttpServer` as instance variables.
- Adding `listen` method which would start listening for http requests. I think it's not good to listen directly from constructor.

**Comment by fvovan** [comment]
agree about separate method for lister but would call it `start` as did for WS server.
and what's the reason to move `router` to instance variable?

**Comment by fvovan** [comment]
@tomaszdurka please review

**Comment by vogdb** [comment]
So, what notation do we use for our dependencies? `"minilog": "^2.1.0"` or `"minilog": "~2.1.0"`?

**Comment by vogdb** [comment]
Can we optimize logger form this:
```js
var Logger = require('./logger');

var log = new Logger(config.get('app.logPath')).getLog();
```
to this
```js
var log = require('./logger');
```
?

Include somehow log instantiation and configuration into './logger' itself.

**Comment by fvovan** [comment]
>So, what notation do we use for our dependencies? "minilog": "^2.1.0" or "minilog": "~2.1.0"?

It's just default one now. AFAIK they mean different things.

**Comment by vogdb** [comment]
> It's just default one now. AFAIK they mean different things. 

Yes. That is why asked :)

**Comment by fvovan** [comment]
but why not leave default values? Is it bad practice in nodejs?
> Can we optimize logger form this?

Then it will be tightly coupled with Application and its Config.

**Comment by vogdb** [comment]
> Then it will be tightly coupled with Application and its Config. 

I'm not sure that it is bad under this environment.

**Comment by vogdb** [comment]
>> Then it will be tightly coupled with Application and its Config. 

> I'm not sure that it is bad under this environment.

It's just we have to create `Logger` in every file where we want to use logs and that I don't like

**Comment by tomaszdurka** [comment]
@fvovan let's pass logger as dependency injection, wdyt?

**Comment by fvovan** [comment]
@tomaszdurka  is done, but no sure if it is more beautiful now.
probably it will be messy every time when you need for logger to debug something you will have to change constructor signature (plus assigning it to instance variable) , instead of just adding 1 line to beginning of the file. but most likely it is okay

**Comment by tomaszdurka** [comment]
Not sure if prettier, but imo better. It's kinda strange as we anyway create all other dependencies inside constructor. We should rather call it `logger` instead of `log`.

**Comment by fvovan** [comment]
true, my concerns were only for "temporary usage". That's why it's a quite common practice to make a Logger as a singleton or  inject it via traits.

**Comment by vogdb** [comment]
@tomaszdurka why DI for logger? O_o Do you want to pass it to all classes as DI? I think we also need to use it in `Auth` class. And you want to have as DI there? In pulsar projects we never used it as DI. Why now?

**Comment by tomaszdurka** [comment]
Yes, I already see problems with this. Probably it's not so necessary especially in languages like Javascript.

**Comment by tomaszdurka** [comment]
Oh reasons are simple - my point of view changed/improved :) But I agree that it will be messy with passing Logger all way down.

**Comment by vogdb** [comment]
as always

**Comment by tomaszdurka** [comment]
@vogdb I think you are right, I've reverted changes in #9

**Pull request merged**